src/core/evaluator.js

```diff
@@ -2421,7 +2426,19 @@ class PartialEvaluator {
         rotate += currentTransform[1] > 0 ? 90 : 270;
       }
 
-      if (rotate !== 0) {
+      if (rotate === -1) {
+        // This is not a 0, 90, 180, 270 rotation so:
+        //  - remove the scale factor from the matrix to get a rotation matrix
+        //  - apply the inverse (which is the transposed) to the positions
+        // and we can then compare positions of the glyphes to detect
+        // a whitespace.
+        [posX, posY] = applyInverseRotation(posX, posY, currentTransform);
+        [lastPosX, lastPosY] = applyInverseRotation(
+          lastPosX,
+          lastPosY,
+          textContentItem.prevTransform
+        );
```

Comment:
Can't we just add this as the `default` branch in the `switch` below, since that should be a tiny bit more efficient overall?